sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/RntbdTransportClient.java

```diff
@@ -541,6 +544,7 @@ private Options(final ConnectionPolicy connectionPolicy) {
             this.tcpKeepIntvl = 1; // Configuration for EpollChannelOption.TCP_KEEPINTVL
             this.tcpKeepIdle = 30; // Configuration for EpollChannelOption.TCP_KEEPIDLE
             this.sslHandshakeTimeoutMinDuration = Duration.ofSeconds(5);
+            this.transientTimeoutDetectionThreshold = 3;
```

Comment:
may be add some java docs explaining internally how this works. or even comments.

Reply:
added more comments